nix/tools/tests.nix

```diff
@@ -105,14 +129,18 @@ let
 
         # collect all tests
         HPCTIXFILE="$tmpdir"/io.tix \
-        ${withTools.latest} ${cabal-install}/bin/cabal v2-exec ${devCabalOptions} \
+          ${withTools.latest} ${cabal-install}/bin/cabal v2-exec ${devCabalOptions} \
           ${ioTestPython}/bin/pytest -- -v test/io-tests
-          
+
         HPCTIXFILE="$tmpdir"/spec.tix \
-        ${withTools.latest} ${cabal-install}/bin/cabal v2-test ${devCabalOptions}
+          ${withTools.latest} ${cabal-install}/bin/cabal v2-run ${devCabalOptions} test:spec
+
+        HPCTIXFILE="$tmpdir"/spec-querycost.tix \
+          ${withTools.latest} ${cabal-install}/bin/cabal v2-run ${devCabalOptions} test:spec-querycost
 
         # collect all the tix files
-        ${ghc}/bin/hpc sum  --union --exclude=Paths_postgrest --output="$tmpdir"/tests.tix "$tmpdir"/io*.tix "$tmpdir"/spec.tix
+        ${ghc}/bin/hpc sum  --union --exclude=Paths_postgrest --output="$tmpdir"/tests.tix \
+          "$tmpdir"/io*.tix "$tmpdir"/spec.tix "$tmpdir"/spec-querycost.tix
```

Comment:
Should we add the doctests to coverage collection, too?

I see you opted for not including them right now, any specific reasons to do so?

Reply:
Bad news: https://downloads.haskell.org/~ghc/8.10.7/docs/html/users_guide/profiling.html#caveats-and-shortcomings-of-haskell-program-coverage

GHCi does not support hpc, and doctest is built on GHCi - so we won't be able to get coverage from `doctest` in its current incarnation